### DISCUSSION TITLE
InputMixer help please

### DISCUSSION THREAD

**Discussion by leelx78** [pschatzmann/arduino-audio-tools#833] [General]
Hi @pschatzmann I'm struggling with InputMixer for a while now so hopefully you can give me a hint here.

This works fine, as per your examples:
```
  I2SStream i2s_stream; //initialized later

...
  info.channels = channels;
  info.sample_rate = sample_rate;
  info.bits_per_sample = 16;

  // Setup sine wave
  sineWave1.begin(channels, sample_rate, N_B4);
  sineWave2.begin(channels, sample_rate, N_E4);

  mixer.add(sound1);
  mixer.add(sound2);
  mixer.begin(info);
  ...
  ```
  
 And the mixer then is piped to I2S. In this case I hear 2 notes and all plays well. But then let's imagine you have 
  ```
  RingBufferStream buffer; //init later
  
  [network] --> SBC Data --> SBC Decoder --> RingBufferStream
  ```
  
If I copy the RingBufferStream directly into I2SStream it works fine but if I add it to the mixer it stops after a split second and nothing happens anymore, all is stuck, e.g.
  
  ```
...
  mixer.add(sound1);
  mixer.add(sound2);
  mixer.add(buffer);
  mixer.begin(info);
```

Why is this? Is it because "buffer" is not continuously generated but comes in in chunks? 

How to handle mixing of sources coming in from the network or Bluetooth, i.e. not continuous? In another post you mentioned synchronisation but in principle wouldn't RingBufferStream take care of that?

Thanks!

**Comment by pschatzmann** [comment]
Are you sure that you are not doing any multitasking w/o task synchronization ?

**Comment by leelx78** [comment]
Sorry @pschatzmann my bad, I found a bug. Working now thanks!
